### PR TITLE
Mongoose cs

### PR DESCRIPTION
**Pull request by C-Stein**
In progress... Working on MEAN milestone exercises. I anticipate that mongo and mongoose will be required for all, while authentication/redis and hosting on mlabs will be optional for students who plan to use mongo for their backend capstones.

### CODE REVIEW

File: 04-getting-mean-with-it/exercises/02-mongoose.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+###Using your mongoose models to write queries
+
+Assuming you have amodel for 'Student', you can write a query to find all students by writing:
+```
+  Student
+  .find()
+  .then(students => console.log({students}))
+  .catch(err)
```
Comment (JoeShep) [inline 04-getting-mean-with-it/exercises/02-mongoose.md]: Are we destructuring students.students in the console.log?

File: 04-getting-mean-with-it/exercises/02-mongoose.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+##Exercise
+
+Go big or go home. Your exercise is to clone the repo at https://github.com/C-Stein/mean-mongoose-zoo and write a server for it using Mongoose. You will need to create
+
+- an express server connected to a mongo database
+- an Animal model created with mongoose
+- get, post, patch, and delete routes that connect to the angular front end provided and interact with your mongo database
+- don't be afraid to check out the mongoose docs and find the best possible query methods for your app
+
+helpful hint: incllude the following middleware if you run into issues with CORS headers
```
Comment (JoeShep) [inline 04-getting-mean-with-it/exercises/02-mongoose.md]: change 'inllude' to 'include'

File: 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# Hosting your mongo database on mlab
+
+###Getting mlab set up
+
+1. Start by visitng mlab.com and creating an account. Log in and click on "create new" button next to MongoDB Deployments.
+1. Select your cloud provider, or ust leave it set on the default. This selection won't matter much for our purposes.
```
Comment (JoeShep) [inline 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md]: change 'ust' to 'just'

File: 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+1. Choose whichever  plan you want, but the sngle-node sandbox plan is free.
+1. give your database a name using lowercase letters.
+1. Click on the name of your newly created databse to go to the detail page
+1. On this detail page, Click on the "users" button, and then "add databasae user"
+1. Create a databasae username and password. Try to make them something you will remember, but DO NOT make them the same as the username and password you used to sign up fpr mlab.
```
Comment (JoeShep) [inline 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md]: change 'fpr mlab' to 'for mlab'

File: 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+1. Now that you have a user, you can substitute your username and password (for the databse) in the line
```
Comment (JoeShep) [inline 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md]: change 'databse' to 'database'

File: 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+module.exports.connect = () => mongoose.connect(MONGODB_URL)
+```
+
+### Hiding your authentication info
+you: Hooray it works!
+me: No! It's not secure.
+
+DO NOT push your mlab database username and password to github. Instead, it's better practice to create a javascript file called "authTokens.js" or something like that. You can the gitignore that file (keeping those tokens secret). Your authTokens file can then look something like:
+```
```
Comment (JoeShep) [inline 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md]: change 'the gitignore' to 'then gitignore'

File: 04-getting-mean-with-it/exercises/02-mongoose.md (hunk 1 above)
Comment (C-Stein) [inline 04-getting-mean-with-it/exercises/02-mongoose.md]: I don't know what we're doing- I shall fix

File: 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md (hunk 5 above)
Comment (C-Stein) [inline 04-getting-mean-with-it/exercises/04-hosting-with-mlabs.md]: it's not right to have so many a's in one word

File: 04-getting-mean-with-it/exercises/02-mongoose.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+//...
+mongoose.connect(MONGODB_URL)
+  .then(() => {
+    app.listen(PORT, () => {
+    console.log(`Hey, I'm listening on port ${PORT}`);
+    })
+  })
+  .catch(console.error)
+```
```
Comment (C-Stein) [inline 04-getting-mean-with-it/exercises/02-mongoose.md]: yes, the same express port, so no conflict. I added the `const PORT = process.env.PORT || 3000` to the example. IS that enough context?

**Comment by JoeShep** [comment]
👍

**Pull request merged**